### PR TITLE
feat: all-in devenv

### PR DESCRIPTION
**Pull request by j8xixo12**
As title, me and @chestercheng refined devenv and it should work properly.
And @chestercheng seems not a member of this group, should we invite him?

### CODE REVIEW

**Review by tai271828 — COMMENTED** [review]
@j8xixo12   would you mind elaborating a bit more "how to test this pull request"?

**Comment by yungyuc** [comment]
> As title, me and @chestercheng refined devenv and it should work properly.
> And @chestercheng seems not a member of this group, should we invite him?

I had invited him.  He hasn't responded the invite.  @chenstercheng could you take a look?

File: bin/devenv [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+add   - create a new devenv environment
+use   - select an environment to use
+del   - delete an environment directory
+off   - deactivate environment
+build - build applications"
+else
+  display -e "Unrecognized command line argument: '${cmd}'"
+fi
+
+# vim: set et nu nobomb fenc=utf8 ft=bash ff=unix sw=2 ts=2:
```
Comment (yungyuc) [inline bin/devenv:28]: In the future, we should document how we like to write the modelines.

**Comment by yungyuc** [comment]
> @j8xixo12 would you mind elaborating a bit more "how to test this pull request"?

I think it's good to add comments for testing.  Having said that, at the current stage, I guess there's little to test.  It's probably safe to postpone testing a little bit.

**Comment by yungyuc** [comment]
The code looks good to me.  I'll merge once @chestercheng responds.

**Comment by chestercheng** [comment]
> > As title, me and @chestercheng refined devenv and it should work properly.
> > And @chestercheng seems not a member of this group, should we invite him?
> 
> I had invited him. He hasn't responded the invite. @chenstercheng could you take a look?

@yungyuc I didn't see the invitation (even in spam), can you resend it?

**Comment by yungyuc** [comment]
> > > As title, me and @chestercheng refined devenv and it should work properly.
> > > And @chestercheng seems not a member of this group, should we invite him?
> > 
> > 
> > I had invited him. He hasn't responded the invite. @chenstercheng could you take a look?
> 
> @yungyuc I didn't see the invitation (even in spam), can you resend it?

Done.  Can you check again?

**Comment by chestercheng** [comment]
Got it, I'm joined !

**Comment by tai271828** [comment]
> > @j8xixo12 would you mind elaborating a bit more "how to test this pull request"?
> 
> I think it's good to add comments for testing. Having said that, at the current stage, I guess there's little to test. It's probably safe to postpone testing a little bit.

+1 to the current stage it's ok to land the refactoring code without much test coverage.

Sorry for my earlier unclear question. What I meant by "test" is more like "usage" actually. I am not sure the usage and expected result changes or not. I misunderstood "refined" and read it as "redefined", so my gut feeling shows something changed. :p Thanks @j8xixo12  @chestercheng for the effort!

**Pull request merged**